lib/core/src/model.rs

```diff
@@ -197,28 +197,37 @@ pub struct ConnectRequest {
 /// An argument when calling [crate::sdk::LiquidSdk::prepare_receive_payment].
 #[derive(Debug, Serialize)]
 pub struct PrepareReceivePaymentRequest {
-    pub payer_amount_sat: u64,
+    pub payer_amount_sat: Option<u64>,
+    pub use_lightning: bool,
```

Comment:
Doesn't this complicate things? It's now a 2x2 matrix of possible inputs, where only 3 combinations are valid. Also the amount range should only be validated when using swaps.

No strong opinion, but IMO it would be clearer to the caller (and easier for us to validate) if we ask for

```rust
pub enum PrepareReceivePaymentRequest {
    Lightning(payer_amount_sat: u64),
    Onchain(payer_amount_sat: Option<u64>)
}
```

Reply:
To expand on Roei's comment above:

1. is IMO more aligned with how we have two methods for payment: `pay_onchain` and `send_payment`, i.e. we don't cover both payment modalities in a single method. Overall this would result in three receive variants: `receive_payment`,  `receive_onchain` and `receive_liquid_payment`.

2. is possible because every swap we create also includes a BIP21 fallback encoded in a routing hint. If we return both, we make it easy for the caller to decide what to show the end-user ("pay this invoice" vs "pay to onchain Liquid address"). The one downside here is it's more complicated to communicate the fees, as it depends if the payer chooses the invoice or the BIP21.